src/app/GUI.py

```diff
@@ -85,28 +85,57 @@ def init(self):
     def _mapiraj(self, x: float, y: float) -> tuple[int]:
         return int(self.sirina * x), int(self.visina * y)
 
-    def _omejitev_pozicije(self):
+    # TODO: da te dve funkciji implementiraš na način da ne
+    #  boš spreminjal nobene stvari iz domenskega prostora.
+    def _omejitev_pozicije_ladje(self):
         sirina = 1 - self.vesolje.ladja.velikost_x
         if self.vesolje.ladja.x >= sirina:
-            self.vesolje.ladja.x = sirina
+            self.key_enabled = False
+        else:
+            self.key_enabled = True
+
+    # TODO: TO FUNKCIJO MI NI USPELO MODIFICIRAT
+    def _omejitev_pozicije_vesoljcev(self):
+        sirina = 1 - self.vesolje.velikost_vesoljca_x
+        if self.vesolje.vesoljci[-1].x >= sirina:
+            for j in range(len(self.vesolje.vesoljci)):
+                self.vesolje.vesoljci[j].y += self.vesolje.premik_vesoljca_navzdol
+                self.vesolje.vesoljci[j].hitrost *= -1
```

Comment:
To da vesoljec zamenja smer potovanja spada v domenski prostor in ne v app layer!
Za vesoljce velja enaka stvar kot za ladjo ki sem napisal, središče vesoljcev naj bo na sredini širina in višina vesoljcev pa tako kot pri ladji... **Sprogramiraj v domeni katera naj upo[teva širino vesoljcev ko boš hotel zamenjati smer vesoljcev** V guiju pa boš naredil tako da se bo domenski prostor pravilno prikazoval in ubogal domenski prostor.

**Spremembe najprej naredi v domeni in šele ko bo v domeni stvar pravilno delala se spravi na app layer**.